docs/public-api-changes-NIO1-to-NIO2.md

```diff
@@ -10,3 +10,6 @@
 - `EventLoop.makePromise`/`makeSucceededFuture`/`makeFailedFuture` instead of `new*`
 - `SocketAddress.makeAddressResolvingHost(:port:)` instead of
   `SocketAddress.newAddressResolving(host:port:)`
+  - Moved error `ChannelLifecycleError.inappropriateOperationForState` to `ChannelError.inappropriateOperationForState`.
```

Comment:
@Lukasa I think all of them needs to have two whitespaces removed so these are not sub-lists.

Reply:
That'll teach me to write my markdown in Xcode.

Reply:
@normanmaurer Resolved, please re-review.